physics/ozphys_time_vary.F90

```diff
@@ -0,0 +1,165 @@
+    real(kind_phys), intent(inout), dimension(:) :: &
+         ddy          ! Interpolation high index for ozone data
+    character(len=*), intent(out) :: &
+         errmsg       ! CCPP error message
+    integer, intent(out) :: &
+         errflg       ! CCPP error flag
+
+    ! Local
+    integer i,j
+
+    ! Initialize CCPP error handling variables
+    errmsg = ''
+    errflg = 0
+
+    ! Set indices
+    do j=1,nPts
+       jindx2(j) = latsozp + 1
+       do i=1,latsozp
+          if (dlat(j) < oz_lat(i)) then
+             jindx2(j) = i
+             exit
+          endif
+       enddo
+       jindx1(j) = max(jindx2(j)-1,1)
+       jindx2(j) = min(jindx2(j),latsozp)
+       if (jindx2(j) .ne. jindx1(j)) then
+          ddy(j) = (dlat(j) - oz_lat(jindx1(j))) / (oz_lat(jindx2(j)) - oz_lat(jindx1(j)))
+       else
+          ddy(j) = 1.0
+       endif
+    enddo
+    
+  end subroutine ozphys_time_vary_init
+
+! ###########################################################################################
+!> \section arg_table_ozphys_time_vary_timestep_init Argument Table
+!! \htmlinclude ozphys_time_vary_timestep_init.html
+!!
+! ###########################################################################################
+  subroutine ozphys_time_vary_timestep_init(nPts, idate, fhour, jindx1, jindx2, latsozp,    &
+       levozp, oz_coeff, timeoz, ozplin, oz_time, oz_lat, ddy, oz_data, errmsg, errflg)
+    ! Inputs
+    integer, intent(in) :: &
+         nPts,     & ! Horizontal dimension
+         latsozp,  & ! Number of latitudes in ozone data
+         levozp,   & ! Number of vertical layers in ozone data
+         oz_coeff, & ! Number of coefficients in ozone data
+         timeoz      ! Number of times in ozone data
+    integer, intent(in),dimension(:) :: &
+         idate,    & ! Initial date with different size and ordering
+         jindx1,   & ! Interpolation index (low) for ozone
+         jindx2      ! Interpolation index (high) for ozone
+    real(kind_phys), intent(in) :: &
+         fhour       ! Forecast hour
+    real(kind_phys), intent(in), dimension(:) :: &
+         ddy,      & ! Interpolation high index for ozone data
+         oz_lat,   & ! Latitudes for ozone data
+         oz_time     ! Time for ozone data
+    real(kind_phys), intent(in), dimension(:,:,:,:) :: &
+         ozplin      ! Ozone data
+
+    ! Outputs
+    real(kind_phys), intent(inout), dimension(:,:,:) :: &
+         oz_data     ! Ozone forcing data
+    character(len=*), intent(out) :: &
+         errmsg      ! CCPP error message
+    integer,          intent(out) :: &
+         errflg      ! CCPP error flag
+
+    ! Local
+    integer :: idat(8),jdat(8),iday,j,j1,j2,l,nc,n1,n2,jdow,jdoy,&
+         jday,w3kindreal,w3kindint
+    real(kind_phys) :: tem, tx1, tx2, rjday
+    real(kind_dbl_prec) :: rinc(5)
+    real(kind_sngl_prec) :: rinc4(5)
+
+    ! Initialize CCPP error handling variables
+    errmsg = ''
+    errflg = 0
+
+    !
+    idat=0
```

Comment:
This code was moved from ozinterpol. I think that h2ointerpol, aerinterpol, iccninterpol are all similar, so couldn't one write generic subroutines for GFS_phys_time_vary and continue to call in from GFS_phys_time_vary, passing in the appropriate arrays for each of ozone, h2o, iccn, aer? This would eliminate extra interstitial schemes.